topic-controller/README.md

```diff
@@ -17,71 +17,61 @@ And also in the other direction:
+changes are not incompatible (e.g. both changing the same topic config key, but to 
 different values). 
+In the case of incompatible changes, the Kafka configuration wins, and the ConfigMap will 
+be updated to reflect that.
 
 
-## Controller configuration
-
-The controller is configured via a ConfigMap within the K8s/OpenShift cluster.
-
-The data of this ConfigMap supports the following keys:
-
-* `kubernetesMasterUrl`, default: `https://localhost:8443`
-* `kafkaBootstrapServers`, default:`localhost:9092`
-* `zookeeperConnect`, default: `localhost:2181`
-* `zookeeperSessionTimeout` a durection for the timeout on the zookeeper session, default: `2 seconds`
-* `fullReconciliationInterval` a duration for the time between full reconciliations, default: `15 minutes`
-* `reassignThrottle`, the throttle to use when topic updates require topic partition reassignment
-* `reassignVerifyInterval` a duration between executions of the `--verify` stage of partition reassignment, default: `2 minutes`  
-
-The controller watches for changes to the config map and reconfigures itself according.
-
-
-## Controller command line and environment
-
-* `--master-url` (or the `CONTROLLER_K8S_URL` env var if absent from the command line) 
-  the URL of the master apiserver in which to find the topic controller's ConfigMap. 
-* `--config-namespace` (or the `CONTROLLER_K8S_NS` env var if absent from the 
-  command line) the namespace within the master apiserver which contains the topic 
-  controller's ConfigMap.
-* `--config-name` (or the `CONTROLLER_K8S_NAME` env var if absent from the 
-  command line) the name of the config map (with the namespace given by `--config-namespace`)
-  which contains the topic 
-  controller's ConfigMap.
-* `--help` command line help, and exit
-* `--help:config` for help on the topic controller's configuration. 
-
+## Controller environment
 
+The controller is configured from environment variables:
 
-## Possible future directions
+* `TC_CM_LABELS` 
+– The Kubernetes label selector used to identify ConfigMaps to be managed by the controller.
+  Default: `strimzi.io/kind=topic`.  
+* `TC_ZK_SESSION_TIMEOUT`
+– The Zookeeper session timeout. For example `10 seconds`. Default: `20 seconds`.
+* `TC_KF_BOOTSTRAP_SERVERS`
+– The list of Kafka bootstrap servers. Default: `${KAFKA_SERVICE_HOST}:${KAFKA_SERVICE_PORT}` 
+* `TC_ZK_CONNECT`
+– The Zookeeper connection information. Default: `${KAFKA_ZOOKEEPER_SERVICE_HOST}:${KAFKA_ZOOKEEPER_SERVICE_PORT}`.
+* `TC_PERIODIC_INTERVAL`
+– The interval between periodic reconciliations.
 
-* Grow an HTTP REST API for changing topics, and representing them as YAML resources, evolving into 
-  an aggregated K8s apiserver.
\ No newline at end of file
+If the controller configuration needs to be changed the process must be killed and restarted.
+Since the controller is intended to execute within Kubernetes, this can be achieved
+by deleting the pod.
```

Comment:
Maybe it's better saying "scaling down and then scaling up for restarting the pod" ? or something like that ?

Reply:
On the command line the simplest thing to do would be to just delete the pod and let k8s recreate it (at least that's what I do!). Scale down and up sound like more than 1 command, but is it better in some way?

Reply:
Well I was thinking to the OpenShift UI and the things are different. Doing `oc delete pods topic-controller` from the command line, as you said you are just killing the pod; the Deployment which is in charge to "watching" for the pods isn't changed and the replica field is still 1 so that it restarts the pod in order to fulfill the request. Doing that in the OpenShift UI manually with the arrows means modifying the Deployment in terms of replica value (so having 0 or 1, for scaling down-up).